Add coverage for abort/error during an in-flight close

domenic opened the pull request:
Closes https://github.com/whatwg/streams/issues/693.

domenic committed:
Add coverage for abort/error during an in-flight close

Closes https://github.com/whatwg/streams/issues/693.

w3c-bots commented:
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5241/)

w3c-bots commented:
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/215887840)

# Firefox (nightly channel) #
Testing web-platform-tests at revision a00fff852008940791bd55b1eaff712be090f07e
Using browser at version BuildID 20170327110256; SourceStamp 9577ddeaafd85554c2a855f385a87472a089d5c0
Starting 10 test iterations

w3c-bots commented:
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/215887841)

# Chrome (unstable channel) #
Testing web-platform-tests at revision a00fff852008940791bd55b1eaff712be090f07e
Using browser at version 59.0.3047.0 dev
Starting 10 test iterations

tyoshino commented on streams/writable-streams/aborting.js:
```diff
@@ -829,6 +924,87 @@ promise_test(t => {
+  }).then(() => {
+    assert_array_equals(
+        events, ['abortPromise'],
+        'writePromise and writer.closed must not be fulfilled/rejected yet even after writer.abort()');
```
s/writePromise/closePromise/

tyoshino commented on streams/writable-streams/aborting.js:
```diff
@@ -829,6 +924,87 @@ promise_test(t => {
+                        'abortPromise, closePromise and writer.closed must fulfill/reject');
+
+    return Promise.all([
+      closePromise,
```
this can be included in the Promise.all at L982?

tyoshino commented on streams/writable-streams/aborting.js:
```diff
@@ -829,6 +924,87 @@ promise_test(t => {
+    return Promise.all([
+      closePromise,
+      promise_rejects(t, error2, writer.ready,
+                      'writer.ready must be still rejected with the error passed to the controller\'s error method')
```
maybe this too as we do nothing in this block?

tyoshino reviewed (approved):
lgtm

domenic committed:
Response to review feedback

Both here and in https://github.com/w3c/web-platform-tests/pull/5242 and in https://github.com/w3c/web-platform-tests/pull/5224.

domenic commented:
@tyoshino if you could take another look, I believe I addressed all your comments in this review and others. I'd like to be sure before merging though.

tyoshino reviewed (approved):
lgtm!
